unified_planning/model/mixins/fluents_set.py

```diff
@@ -135,8 +135,10 @@ def add_fluent(
             fluent = up.model.fluent.Fluent(
                 fluent_or_name, typename, None, env=self.env, **kwargs
             )
-        if self._has_name_method(fluent.name):
-            raise UPProblemDefinitionError("Name " + fluent.name + " already defined!")
+        if self.has_fluent(fluent.name):
+            raise UPProblemDefinitionError(
+                "Fluent with name " + fluent.name + " already defined!"
+            )
```

Comment:
Why do we need this change?

Reply:
The previous implementation disallowed to have two things with the same name in the UP even if they are non-colliding (e.g. a type and an action).
While this is generally bad practice, it is not forbidden in the PDDL standard and some domains rely on it (in particular, at least one domain in the HTN IPC relies on it).

Maybe we could issue a warning in this case, but he exception would rule out valid, unambiguous, domains.

Reply:
Yes, you're right. I missed this possible ambiguity. An option would be to have a more fine grained check there.

Reply:
I agree that we should have a more fine grained check. I propose to undo these changes in this PR, remove the problematic domains from the tests and open a dedicated issue.

Then we will handle this in a new PR.

Reply:
@arbimo what do you think about my proposal here?

Reply:
I am unsure about this one. In python, we never manipulate the name but instead use the previously constructed python objects, so I do not see any risk of ambiguity there. There I believe it would make sense to remove the check entirely (which would also simplify this error prone part of the code with cross-mixins dependencies).

In the PDDL parser, we can detect and raise the ambiguity at its point of occurrence : when parsing expressions. This would also give the opportunity to provide better error messages.

Reply:
I agree that we should relax these checks, even removing them entirely, but I would do it in a separate PR.
Removing them entirely also means to update the ANML/PDDL writers to handle the possible ambiguities.

Reply:
@arbimo if we relax/remove the checks in a different PR and we undo the changes from here, I think we can merge the PR.

Reply:
Yes, I just pushed the changes, and removed the three impacted domains that now failed to parse.